test/unit/parser_test.rb

```diff
@@ -408,6 +410,58 @@ def test_expression_conditions_attribute
     end
   end
 
+  def test_nested_nots
+    @parser = Parser.new
+    expressions = @parser.parse('Not Test Eq 11')
+    assert !@parser.errors?, @parser.inspect
+    expression = expressions.last
+    assert_equal 11.to_s, expression[:value]
+    assert_equal "Not", expression[:unary]
+    assert_equal "And", expression[:conjunction]
+
+    @parser = Parser.new
+    expressions = @parser.parse('Not (Not Test Eq 11)')
+    assert !@parser.errors?, @parser.inspect
+    expression = expressions.last
+    assert_equal 11.to_s, expression[:value]
+    assert_nil expression[:unary]
+    assert_equal "And", expression[:conjunction]
+
+    @parser = Parser.new
+    expressions = @parser.parse('Not (Not (Not Test Eq 11))')
+    assert !@parser.errors?, @parser.inspect
+    expression = expressions.last
+    assert_equal 11.to_s, expression[:value]
+    assert_equal 'Not', expression[:unary]
+    assert_equal "And", expression[:conjunction]
+
+    @parser = Parser.new
+    expressions = @parser.parse('Not (Not (Not (Not ListPrice Eq 1000 Not ListPrice Eq 2000)))')
+    assert_equal 2, expressions.length
+    exp = expressions[0]
+    assert_equal "And", exp[:conjunction]
+    assert_equal "1000", exp[:value]
+    assert_nil exp[:unary]
+
+    exp = expressions[1]
+    assert_equal "And", exp[:conjunction]
+    assert_equal "2000", exp[:value]
+    assert_nil exp[:unary]
+
+    @parser = Parser.new
+    expressions = @parser.parse('Not (Not (Not ListPrice Eq 1000 Not ListPrice Eq 2000))')
```

Comment:
Added the test, but it would be more accurate to say that `Not` is *never* a conjunction